showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITCrud.java

```diff
@@ -0,0 +1,177 @@
+                    .setHttpTransport(
+                        new NetHttpTransport.Builder().doNotValidateCertificate().build())
+                    .setEndpoint("http://localhost:7469")
+                    .build())
+            .build();
+    httpJsonClient = IdentityClient.create(httpJsonIdentitySettings);
+
+    // Ensure an empty state before each run
+    cleanupData(httpJsonClient);
+  }
+
+  @After
+  public void cleanup() {
+    httpJsonClient.close();
+  }
+
+  private void cleanupData(IdentityClient identityClient) {
+    IdentityClient.ListUsersPagedResponse pagedResponse =
+        identityClient.listUsers(ListUsersRequest.newBuilder().setPageSize(5).build());
+    for (IdentityClient.ListUsersPage listUsersPage : pagedResponse.iteratePages()) {
+      for (User user : listUsersPage.getResponse().getUsersList()) {
+        identityClient.deleteUser(user.getName());
+      }
+    }
+    pagedResponse = httpJsonClient.listUsers(ListUsersRequest.newBuilder().setPageSize(5).build());
+    assertThat(pagedResponse.getPage().getResponse().getUsersList().size()).isEqualTo(0);
+  }
+
+  // Helper method to create a user with the DEFAULT_USER configs. Server returns
+  // a generated name (not username) that is used to identify the individual user
+  // and each test uses the name for the RPC
+  private User createDefaultUser() {
+    return httpJsonClient.createUser(CreateUserRequest.newBuilder().setUser(DEFAULT_USER).build());
+  }
+
+  @Test
+  public void testHttpJson_Create() {
+    User userResponse = createDefaultUser();
+
+    // These properties should be the same
+    assertThat(userResponse.getDisplayName()).isEqualTo(DEFAULT_USER.getDisplayName());
+    assertThat(userResponse.getEmail()).isEqualTo(DEFAULT_USER.getEmail());
+    assertThat(userResponse.getNickname()).isEqualTo(DEFAULT_USER.getNickname());
+    assertThat(userResponse.getHeightFeet()).isEqualTo(DEFAULT_USER.getHeightFeet());
+    assertThat(userResponse.getAge()).isEqualTo(DEFAULT_USER.getAge());
+
+    // Assert that the server populates these fields
+    assertThat(userResponse.getName()).isNotEmpty();
+    assertThat(userResponse.getCreateTime()).isNotNull();
+    assertThat(userResponse.getUpdateTime()).isNotNull();
+    assertThat(userResponse.getEnableNotifications()).isNotNull();
+  }
+
+  @Test
+  public void testHttpJson_Read() {
+    User userResponse = createDefaultUser();
+    // Assert that only one User exists
+    IdentityClient.ListUsersPagedResponse listUsersPagedResponse =
+        httpJsonClient.listUsers(ListUsersRequest.newBuilder().setPageSize(5).build());
+    ListUsersResponse listUsersResponse = listUsersPagedResponse.getPage().getResponse();
+    assertThat(listUsersResponse.getUsersList().size()).isEqualTo(1);
+
+    // Check that the response from both List (pagination) and Get
+    // List Users
+    User listUserResponse = listUsersResponse.getUsers(0);
+    assertThat(listUserResponse).isEqualTo(userResponse);
```

Comment:
Would it be worth testing a scenario where we have more than one User?

Reply:
Sure, I'll add another user in.